data-prepper-plugins/grok-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepperTests.java

```diff
@@ -59,52 +59,59 @@
     @Mock
     private Match match;
 
+    @Mock
+    private Match match2;
```

Comment:
Why are these `2` fields here?

Unless I missed something, it appears that they are not setup differently in the setup method. If that is the case, then you can just re-use the existing fields.

If they are being setup differently, then the best solution would be to use `@Nested` classes.

Reply:
Having the same field names for both won't work as the `Nested` tests also need to use the original fields. I will use some more descriptive names than just adding the `2`

Reply:
While splitting them up like this would make sense, the `match` and `capture` are used in every single test, even the `@Nested` ones that use `match2`. So it makes sense for the top level class to have the tests that don't use `match2`, and the `@Nested` class to have the tests that do use `match2`.